src/lib/evil/unposix/fnmatch.c

```diff
@@ -0,0 +1,271 @@
+                                else                       count += 5;
+                             }
+                        }
+                      else
+                        {
+                           if (j == 0) count += 10;
+                           else
+                             {
+                                if (pattern[j - 1] == '/') count += 10;
+                                else                       count += 4;
+                             }
+                        }
+                   }
+                 else  // PATHNAME
+                   {
+                      if (pattern[j] == '*') count += 5;
+                      else                   count += 4;
+                   }
+              }
+            else if (flags & FNM_PERIOD)
+              {
+                 if (j == 0)  // period at init
+                   {
+                      if (pattern[j] == '*') count += 5;
+                      else                   count += 4;
+                   }
+                 else  // period at other places
+                   {
+                      if (pattern[j] == '*') count += 2;
+                      else                   count++;
+                   }
+              }
+            else  // NORMAL
+              {
+                 if (pattern[j] == '*') count += 2;
+                 else                   count++;
+              }
+         }
+       else if (pattern[j] == '.') count += 4;
+       else                        count++;  // OTHERS
+    }
+
+  int reg_length = pattern_length + count;
+
+  // Translates wildcards to regex
+  char *reg_pattern = (char *)malloc(reg_length * sizeof(char));
+  if (reg_pattern == NULL) exit(ENOMEM);
+  reg_pattern[0] = '^';
+  int i = 1;
+  for (int j = 0; j < pattern_length; ++j)
+    {
+       if (pattern[j] == '\\')
+         {
+            if (flags && FNM_NOESCAPE)
+               reg_pattern[i++] = '/';
+            else
+              {
+                 if (_is_escapable(pattern[j + 1]))
+                   {
+                      reg_pattern[i++] = '\\';
+                      reg_pattern[i++] = pattern[++j];
+                   }
+                 else
+                   {
+                      reg_pattern[i++] = '\\';
+                      reg_pattern[i++] = '/';
+                   }
+              }
+         }
+       else if ((pattern[j] == '*') || (pattern[j] == '?'))
+         {
+            if (flags & FNM_PATHNAME)
+              {
+                 if (flags & FNM_PERIOD)  // PATHNAME + PERIOD
+                   {
+                      if (pattern[j] == '*')
+                        {
+                           if (j == 0)
+                             {
+                                strcpy(reg_pattern + i, "[^\\.][^/]*");
+                                i += 10;
+                             }
+                           else
+                             {
+                                if (pattern[j - 1] == '/')
```

Comment:
Not exactly asking a change, but why you have so much:
```
if () {}
else 
{
  if () {}
  else {}
}
```
Why not making those `else if`'s?
```
if () {}
else if () {}
else {}
```

Reply:
Yeah, that was the way my mind deal with the decision tree and of course it could be rewritten for clarity. Do you think that make it more flat will be easier to understand? I will see how it looks when flatted...

Reply:
For me at least `else if`'s makes it easier for understanding the flow.